From 6da2248f151d19960efa82e657501ec2080b4c13 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Wed, 5 Aug 2015 17:00:09 +0800
Subject: [PATCH] MLK-11324 dma: pxp: create a Kconfig for pxp to hold related
 configs

Move pxp related kernel configs to this new created
Kconfig to make the config info more clear and easier
to management.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/dma/Kconfig     | 22 +---------------------
 drivers/dma/pxp/Kconfig | 21 +++++++++++++++++++++
 2 files changed, 22 insertions(+), 21 deletions(-)

diff --git a/drivers/dma/Kconfig b/drivers/dma/Kconfig
index acdeecd..a7d9809 100644
--- a/drivers/dma/Kconfig
+++ b/drivers/dma/Kconfig
@@ -137,27 +137,7 @@ config MX3_IPU_IRQS
 	  To avoid bloating the irq_desc[] array we allocate a sufficient
 	  number of IRQ slots and map them dynamically to specific sources.
 
-config MXC_PXP_V2
-	bool "MXC PxP V2 support"
-	depends on ARM
-	select DMA_ENGINE
-	help
-          Support the PxP (Pixel Pipeline) on i.MX6 DualLite and i.MX6 SoloLite.
-	  If unsure, select N.
-
-config MXC_PXP_V3
-	bool "MXC PxP V3 support"
-	depends on ARM
-	select DMA_ENGINE
-	help
-	  Support the PxP V3(Pixel Pipeline) on i.MX7D. The PxP V3 supports
-	  more functions than PxP V2, dithering, reagl/-D and etc.
-	  If unsure, select N.
-
-config MXC_PXP_CLIENT_DEVICE
-	bool "MXC PxP Client Device"
-	default y
-	depends on MXC_PXP_V2 || MXC_PXP_V3
+source "drivers/dma/pxp/Kconfig"
 
 config TXX9_DMAC
 	tristate "Toshiba TXx9 SoC DMA support"
diff --git a/drivers/dma/pxp/Kconfig b/drivers/dma/pxp/Kconfig
new file mode 100644
index 0000000..5a280a8
--- /dev/null
+++ b/drivers/dma/pxp/Kconfig
@@ -0,0 +1,21 @@
+config MXC_PXP_V2
+	bool "MXC PxP V2 support"
+	depends on ARM
+	select DMA_ENGINE
+	help
+          Support the PxP (Pixel Pipeline) on i.MX6 DualLite and i.MX6 SoloLite.
+	  If unsure, select N.
+
+config MXC_PXP_V3
+	bool "MXC PxP V3 support"
+	depends on ARM
+	select DMA_ENGINE
+	help
+	  Support the PxP V3(Pixel Pipeline) on i.MX7D. The PxP V3 supports
+	  more functions than PxP V2, dithering, reagl/-D and etc.
+	  If unsure, select N.
+
+config MXC_PXP_CLIENT_DEVICE
+	bool "MXC PxP Client Device"
+	default y
+	depends on MXC_PXP_V2 || MXC_PXP_V3
-- 
1.8.0

